### ISSUE TITLE
MSTest.TestAdapter nuget package 1.2.0 cannot be used.

### ISSUE THREAD

**Issue by rzio** [Microsoft/testfx#383]
## Can't use MSTest.TestAdapter 1.2.0 nuget package
When trying to compile a project (class library, .net 461, any CPU or X64) with the MSTest.TestAdapter 1.2.0 package the compilation fails. (This was suggested in https://github.com/Microsoft/testfx/issues/381)

## Steps to reproduce
* Create new solution.
* Create new class library project:
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>net461</TargetFramework>
  </PropertyGroup>
  <ItemGroup>
    <PackageReference Include="MSTest.TestAdapter" Version="1.2.0" />
  </ItemGroup>
</Project>
```
* Create class:
```C#
using System;
using Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter;

namespace TestExecutor
{
    public class Class1
    {
        public void Run()
        {
            var t = new MSTestDiscoverer();
        }
    }
}
```
* Compile

## Expected behavior
Compilation succeeds

## Actual behavior
Compilation fails with errors
```
Error	CS0234	The type or namespace name 'VisualStudio' does not exist in the namespace 'Microsoft' (are you missing an assembly reference?)	TestExecutor	c:\users\alexrez\source\repos\TestExecutor\Class1.cs	2	Active
Error	CS0246	The type or namespace name 'MSTestDiscoverer' could not be found (are you missing a using directive or an assembly reference?)	TestExecutor	c:\users\alexrez\source\repos\TestExecutor\Class1.cs	10	Active
========== Build: 0 succeeded, 1 failed, 0 up-to-date, 0 skipped ==========
```

**Comment by smadala** [comment]
I had discussed with @rzio offline. 
@rzio  Can you please update the thread with discussion? Share your ask here?

**Comment by rzio** [comment]
Following up on the ask and scenarios:
I'm trying to execute tests in Azure Function. (Those tests will send an HTTP request to a RESTful service, and assert of the response). My aim is to share the test code between the integration tests that are run as a part of the development cycle (during which the request will be sent to a in-proc asp.net server), deployment verification which are run in VSTS and Test-In-Production that monitors the service periodically.

So the asks:

- Change the TestAdapter nuget package, so I could compile my Azure function with against it.
- Add some way to pass configuration between the hosting process and the test code (currently I set an Environment variable at the process scope, but it's fragile and not the best solution to pass complicated configurations)
- Add some more documentation on the web, since I had to basically reverse engineer what's going on by reading the code at github.

Thank you for considering this.
/rzio
